### PR TITLE
Remove deprecated comparison kernels

### LINKED ISSUE #4733: Remove Deprecated Comparison Kernels
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

#4701 deprecated a large number of kernels in arrow-ord

**Describe the solution you'd like**

Translate the existing tests to the new APIs, and remove the legacy kernels.

This should be a fairly mechanical exercise and so marking this as a good first issue

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zinking**
# Which issue does this PR close?

Closes #4733 [Remove Deprecated Comparison Kernels](https://github.com/apache/arrow-rs/issues/4733#top)

# Rationale for this change
 

# What changes are included in this PR?

this is the first part of the change, which updates all tests of the old kernel tests

# Are there any user-facing changes?

NO

### CODE REVIEW

**Comment by zinking** [comment]
@tustvold like you suggested, this PR updates all tests in place for the convenience of review. take a look when you have time.

File: arrow-ord/src/comparison.rs [hunk 1]
```diff
@@ -1548,79 +1547,228 @@ mod tests {
...
-    #[test]
-    fn test_primitive_array_eq() {
-        cmp_i64!(
-            eq,
-            eq_dyn,
-            vec![8, 8, 8, 8, 8, 8, 8, 8, 8, 8],
-            vec![6, 7, 8, 9, 10, 6, 7, 8, 9, 10],
-            vec![false, false, true, false, false, false, false, true, false, false]
-        );
+    macro_rules! test_array_scalar_op {
```
Comment (tustvold) [inline arrow-ord/src/comparison.rs:1550]: I know there were test macros before, but perhaps we could take the opportunity to remove them. I find it really hard to follow what these tests are doing now...

File: arrow-ord/src/comparison.rs [hunk 2]
```diff
@@ -1641,403 +1789,242 @@ mod tests {
...
-            vec![Some(true), Some(true), Some(false), Some(false), None, Some(false)]
-                .into();
-
-        let res: Vec<Option<bool>> = lt_bool(&a, &b).unwrap().iter().collect();
-
-        assert_eq!(
-            res,
-            vec![Some(false), Some(true), Some(false), Some(false), None, None]
-        )
+        test_boolean_array_array_op!(crate::cmp::lt, "lt")
```
Comment (tustvold) [inline arrow-ord/src/comparison.rs:1813]: I think it would be much easier to follow this if it simply was something like



```
let a = BooleanArray::from(vec![...]);
let b = BooleanArray::from(vec![...]);
let expected = BooleanArray::from(vec![...]);
assert_eq(lt(&a, &b).unwrap(), expected);
```

**Review by tustvold — COMMENTED** [review]
Thank you for working on this, I wonder if instead of adding more macros it might make the code easier to follow if we just didn't use them?

**Comment by zinking** [comment]
I'm okay with removing macros as long as codes are not duplicated everywhere.

I did try to come up with generics to make a corresponding scalar in `test_primitive_dyn_scalar` didn't manage to do it.

**Comment by tustvold** [comment]
Marking this as a draft to make clear it isn't awaiting review, feel free to unmark when you would like me to take another look
